lib/health_cards/health_card.rb

```diff
@@ -1,214 +1,58 @@
-        jws = JWS.from_jws(jws, public_key: public_key, key: key)
-        from_payload(jws.payload)
-      end
-
-      # Create a HealthCard from a compressed payload
-      # @param payload [String]
-      # @return [HealthCards::HealthCard]
-      def from_payload(payload)
-        json = decompress_payload(payload)
-        bundle_hash = json.dig('vc', 'credentialSubject', 'fhirBundle')
-
-        raise HealthCards::InvalidCredentialError unless bundle_hash
+    extend Forwardable
 
-        bundle = FHIR::Bundle.new(bundle_hash)
-        new(issuer: json['iss'], bundle: bundle)
-      end
-
-      # Decompress an arbitrary payload, useful for debugging
-      # @param payload [String] compressed payload
-      # @return [Hash] Hash built from JSON contents of payload
-      def decompress_payload(payload)
-        inf = Zlib::Inflate.new(-Zlib::MAX_WBITS).inflate(payload)
-        JSON.parse(inf)
-      end
+    attr_reader :jws
 
-      # Compress an arbitrary payload, useful for debugging
-      # @param payload [Object] Any object that responds to to_s
-      # @return A compressed version of that payload parameter
-      def compress_payload(payload)
-        Zlib::Deflate.new(nil, -Zlib::MAX_WBITS).deflate(payload.to_s, Zlib::FINISH)
-      end
+    def_delegator :@jws, :verify
+    def_delegator :@qr_codes, :code_by_ordinal
+    def_delegators :@payload, :bundle, :issuer
 
-      # Sets/Gets the fhir version that will be passed through to the credential created by an instnace of
-      # this HealthCard (sub)class
-      # @param ver [String] FHIR Version supported by this HealthCard (sub)class. Leaving this param out
-      # will only return the current value
-      # value (used as a getter)
-      # @return [String] Current FHIR version supported
-      def fhir_version(ver = nil)
-        if @fhir_version.nil? && ver.nil?
-          @fhir_version = superclass.fhir_version unless self == HealthCards::HealthCard
-        elsif ver
-          @fhir_version = ver
-        end
-        @fhir_version
-      end
+    def initialize(jws)
+      @jws = jws.is_a?(String) ? JWS.from_jws(jws) : jws
+      @payload = Payload.from_payload(@jws.payload)
+      @qr_codes = QRCodes.from_jws(@jws)
     end
 
-    fhir_version '4.0.1'
-
-    additional_types 'https://smarthealth.cards#health-card'
-
-    allow type: FHIR::Meta, attributes: %w[security]
-
-    disallow attributes: %w[id text]
-    disallow type: FHIR::CodeableConcept, attributes: %w[text]
-    disallow type: FHIR::Coding, attributes: %w[display]
-
-    # Create a HealthCard
-    #
-    # @param bundle [FHIR::Bundle] VerifiableCredential containing a fhir bundle
-    # @param issuer [String] The url from the Issuer of the HealthCard
-    def initialize(bundle:, issuer: nil)
-      raise InvalidPayloadError unless bundle.is_a?(FHIR::Bundle) # && bundle.valid?
-
-      @issuer = issuer
-      @bundle = bundle
+    def credential
+      { verifiableCredential: [@jws.to_s] }
```

Comment:
```suggestion
    def download
      Exporter.download(@jws)
```

This is already implemented in [Exporter where it is called `download`](https://github.com/dvci/health_cards/blob/bfb1d66811662270f777222ae50aeb3502a876d4/lib/health_cards/exporter.rb#L10-L12) since it's actually the [File Download Format](https://spec.smarthealth.cards/#via-file-download).

We should be consistent with the naming. I don't really like the `download` name either though so I'm up for suggestions (`file_download_json`?)